### PR TITLE
Sign messages by Trezor

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/19219

- Added support for signing messages with Trezor

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Call eth_sign with hardware transactions

### CODE REVIEW

File: components/brave_wallet_ui/trezor/trezor.ts [hunk 1]
```diff
@@ -58,3 +61,13 @@ addTrezorCommandHandler(TrezorCommand.SignTransaction, (command: SignTransaction
     })
   })
 })
+
+addTrezorCommandHandler(TrezorCommand.SignMessage, (command: SignMessageCommand, source: Window): Promise<SignMessageResponsePayload> => {
+  return new Promise(async (resolve) => {
+    console.log(command.payload)
```
Comment (darkdh) [inline components/brave_wallet_ui/trezor/trezor.ts:67]: this should be removed

File: components/brave_wallet_ui/trezor/trezor.ts [hunk 2]
```diff
@@ -58,3 +61,13 @@ addTrezorCommandHandler(TrezorCommand.SignTransaction, (command: SignTransaction
...
+    TrezorConnect.ethereumSignMessage(command.payload).then((result: SignMessageResponse) => {
+      console.log(result)
```
Comment (darkdh) [inline components/brave_wallet_ui/trezor/trezor.ts:69]: this should be removed

**Review by darkdh — APPROVED** [review]

File: components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts [hunk 3]
```diff
@@ -247,34 +250,35 @@ handler.on(PanelActions.signMessageProcessed.getType(), async (store: Store, pay
...
-        })
+  if (!hardwareAccount || !hardwareAccount.hardware) {
+    const braveWalletService = apiProxy.braveWalletService
+    braveWalletService.notifySignMessageHardwareRequestProcessed(false, messageData.id,
+      '', getLocale('braveWalletHardwareAccountNotFound'))
+    const signMessageRequest = await getPendingSignMessageRequest()
+    if (signMessageRequest) {
+      store.dispatch(PanelActions.signMessage(signMessageRequest))
+      return
     }
```
Comment (petemill) [inline components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts:262]: Just a question (and maybe warrants a code comment in a follow-up): What does this do? Attempt to re-sign the request again? Or sign the next request if there is one?

File: components/resources/wallet_strings.grdp [hunk 4]
```diff
@@ -346,4 +346,6 @@
   <message name="IDS_BRAVE_WALLET_HARDWARE_TRANSACTION_NOT_FOUND_ERROR" desc="Unable to find hardware transaction during signing process">Unable to find hardware transaction, please try again</message>
   <message name="IDS_BRAVE_WALLET_HARDWARE_TRANSACTION_DEVICE_ERROR" desc="Received error from hardware device during signing process">Received error from hardware device, probably you need to confirm the transaction on your device</message>
   <message name="IDS_BRAVE_WALLET_HARDWARE_TRANSACTION_NO_MESSAGE_TO_SIGN_ERROR" desc="Unable to get raw messagse to sign for hardware transaction during signing process">Error processing the transaction, please try again</message>
+  <message name="IDS_BRAVE_WALLET_HARDWARE_SIGN_MESSAGE_ERROR" desc="Unable to sign personal messagse on hardware device">Error signing message, please try again</message>
+  <message name="IDS_BRAVE_WALLET_HARDWARE_UNKNOWN_KEYRING_ERROR" desc="Error when requested unknown keyring">Error signing message, unknown keyring requeted, please try again</message>
```
Comment (petemill) [inline components/resources/wallet_strings.grdp:350]: spelling: requeted = requested

**Review by petemill — APPROVED** [review]
Looks good. Please fix string spelling before merge, anything else is just a comment

File: components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts (hunk 3 above)
Comment (spylogsster) [reply components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts:262]: sign next if there is one

**Pull request merged**